tests/workers/test_ecs_worker.py

```diff
@@ -2016,6 +2016,48 @@ async def test_user_defined_environment_variables_in_task_definition_template(
     ]
 
 
+@pytest.mark.usefixtures("ecs_mocks")
+async def test_user_defined_capacity_provider_strategy(
+    aws_credentials: AwsCredentials, flow_run: FlowRun
+):
+    configuration = await construct_configuration_with_job_template(
+        template_overrides=dict(
+            task_run_request={
+                "capacityProviderStrategy": [
+                    {"base": 0, "weight": 1, "capacityProvider": "r6i.large"},
+                ]
+            },
+        ),
+        aws_credentials=aws_credentials,
+    )
+
+    assert "launchType" not in configuration.task_run_request
```

Comment:
do we want to assert that things work when `launchType` is omitted altogether or when it is passed with a value of `None`?
> Change the validation so that launch_type=None is okay when using publish_as_work_pool.


I don't have great context here, just asking 🙂

Reply:
I think we want to test that it is getting popped off from the `task_run_request` because there isn't a way for us to pass `launch_type`. This does make me think, should we let users be able to pass it as long as we create an `ECSvariables` field for `capacityProviderStrategy`?